### PR TITLE
Add method for solve the movement problem at the switch mode

### LINKED ISSUE #118: Stop reality body movement when switching to noclip

### PR DESCRIPTION
**Pull request by stefanofossati**
The problem seems to be fixed:

- Add a method in the realitymovement script that is called when the switch from reality mode to noclip mode happend

### CODE REVIEW

**Review by spig95 — DISMISSED** [review]
Methods name make it really clear, and it works!

**Review by TizioMaurizio — CHANGES_REQUESTED** [review]
I suggest to make it so that the reality body is unaffected by forces while in noclip mode to account for situations like this that would cause respawn loops, the player shouldn't move anyway in noclip
Untitled

File: Assets/Code/Scripts/NoclipRealityManagement/RealityMovementCalibration.cs [hunk 1]
```diff
@@ -160,6 +161,12 @@ public MovementState GetState()
     {
         return _state;
     }
+
```
Comment (TizioMaurizio) [inline Assets/Code/Scripts/NoclipRealityManagement/RealityMovementCalibration.cs]:
```suggestion
public void toggleKinematic(bool toggle)
    {
        _rigidbody.isKinematic = toggle;
    }
```

File: Assets/Code/Scripts/PlayerManagement/CameraManager.cs [hunk 2]
```diff
@@ -48,8 +50,10 @@ public void SwitchCamera()
         
         if (isNoclipEnabled) //When the switch from reality mode to noclip mode happened
         {
+            
             _noclipMovement.SetPositionAndRotation(_realPlayer.transform.position, _realPlayerCamera.transform.rotation); //Set the noclip position in the realBody position
             _noclipMouseLook.CopyRotationCoordinates(_realMouseLook);
+            _realityMovement.ResetMovementAtSwitch();
```
Comment (TizioMaurizio) [inline Assets/Code/Scripts/PlayerManagement/CameraManager.cs]:
```suggestion
            _realityMovement.toggleKinematic(isNoclipEnabled);
```

File: Assets/Code/Scripts/PlayerManagement/CameraManager.cs [hunk 3]
```diff
@@ -48,8 +50,10 @@ public void SwitchCamera()
...
         }
         else
         {
```
Comment (TizioMaurizio) [inline Assets/Code/Scripts/PlayerManagement/CameraManager.cs:59]:
```suggestion
        {
            _realityMovement.toggleKinematic(isNoclipEnabled);
```

**Review by TizioMaurizio — DISMISSED** [review]

**Review by TizioMaurizio — APPROVED** [review]

**Pull request merged**